java/src/datahike/java/Datahike.java

```diff
@@ -0,0 +1,153 @@
+        deleteDatabaseFn.invoke(uri);
+    }
+
+    public static void createDatabase(String uri) {
+        createDatabaseFn.invoke(uri);
+    }
+
+    public static void createDatabase(String uri, Object... args) {
+        List argsCopy = new ArrayList(Arrays.asList(args));
+        argsCopy.add(0, uri);
+        createDatabaseFn.applyTo(RT.seq(argsCopy));
+    }
+
+    public static Object connect(String uri) {
+        return connectFn.invoke(uri);
+    }
+
+    public static Object db(Object conn) {
+        return dbFn.invoke(conn);
+    }
+
+    public static Set<PersistentVector> q(String query, Object... inputs) {
+        List argsCopy = new ArrayList(Arrays.asList(inputs));
+        argsCopy.add(0, Clojure.read(query));
+        return (Set<PersistentVector>)qFn.applyTo(RT.seq(argsCopy));
+    }
+
+    public static Object transact(Object conn, PersistentVector txData) {
+        return transactFn.invoke(conn, txData);
+    }
+
+    /** dConn: the dereferenced conn object */
+    public static Object history(Object dConn) { return historyFn.invoke(dConn); };
+
+    public static Object asOf(Object dConn, Date date) {
+        return asOfFn.invoke(dConn, date);
+    }
+
+    public static Object since(Object dConn, Date date) {
+        return sinceFn.invoke(dConn, date);
+    }
+
+    public static Map pull(Object dConn, String selector, int eid) {
+        return (Map) pullFn.invoke(dConn, Clojure.read(selector), eid);
+    }
+
+    public static Map pull(Object dConn, String selector, PersistentVector eid) {
+        return (Map) pullFn.invoke(dConn, Clojure.read(selector), eid);
+    }
+
+
+    public static List pullMany(Object dConn, String selector, PersistentVector eids) {
+        return (List) pullManyFn.invoke(dConn, Clojure.read(selector), eids);
+    }
+
+    public static void release(Object conn) {
+        releaseFn.invoke(conn);
+    }
+
+    public static List seekdatoms(Object dConn, Keyword index, Object c1) {
+        return (List)seekDatomsFn.invoke(dConn, index, c1);
+    }
+
+    public static List seekdatoms(Object dConn, Keyword index, Object c1, Object c2) {
+        return (List)seekDatomsFn.invoke(dConn, index, c1, c2);
+    }
+
+    public static List seekdatoms(Object dConn, Keyword index, Object c1, Object c2, Object c3) {
+        return (List)seekDatomsFn.invoke(dConn, index, c1, c2, c3);
+    }
+
+    public static List seekdatoms(Object dConn, Keyword index, Object c1, Object c2, Object c3, Object c4) {
+        return (List)seekDatomsFn.invoke(dConn, index, c1, c2, c3, c4);
+    }
+
+    public static Long tempId(Keyword k) {
+        return (Long) tempIdFn.invoke(k);
+    }
+
+    public static Long tempId(Keyword k, Long i) {
+        return (Long)tempIdFn.invoke(k, i);
+    }
+
+    public static IEntity entity(Object dConn, int eid) {
+        return (IEntity)entityFn.invoke(dConn, eid);
+    }
+
+    // TODO: this would also benefit from an Entity interface
```

Comment:
What would the Entity interface work here?

Reply:
This is how the method is redefined using the IEntity interface:

public static Object entityDb(IEntity entity) {